bindings/python3/zenroom/zenroom.py

```diff
@@ -33,55 +24,29 @@ def __post_init__(self):
             self.result = None
 
 
-def _libzenroom_exec(language, script, conf, keys, data):
-    """Call either zenroom_exec of zencode_exec function depending on language
-    argument."""
-    def _char_p(x):
-        return ct.c_char_p(None if x is None else x.encode())
-
-    case_apply(
-        language,
-        {_ZenLanguage.lua: _LIBZENROOM.zenroom_exec,
-         _ZenLanguage.zencode: _LIBZENROOM.zencode_exec},
-        _char_p(script),
-        _char_p(conf),
-        _char_p(keys),
-        _char_p(data)
-    )
-
+def _char_p(x):
+    return ct.c_char_p(None if x is None else x.encode())
 
-def lua_exec(script, conf=None, keys=None, data=None):
-    """Execute Lua in Zenroom environment"""
-    _libzenroom_exec(_ZenLanguage.lua, script, conf, keys, data)
 
-
-def zencode_exec(script, conf=None, keys=None, data=None):
-    """Execute Zencode in Zenroom environment"""
-    _libzenroom_exec(_ZenLanguage.zencode, script, conf, keys, data)
-
-
-def _zenroom_call(language, script, conf, keys, data):
+def _apply_call(call, script, conf, keys, data):
```

Comment:
`_apply_call` -- seems to be strangely named to me. Apply means apply a function to arglist which essentially same as call a function.  Maybe just `_apply`?

Reply:
And also the `call` argument shouldn't it be `function`?
When I see `call(arg1, arg2, arg3)` I instantly think of `arg1(arg2, arg3)` or something like that.

Reply:
> _apply_call -- seems to be strangely named to me. Apply means apply a function to arglist which essentially same as call a function. Maybe just _apply?

I prefer a more verbose naming style, that's what I have used `_apply_call` instead of `_apply`:

```
def zenroom_exec(script, conf=None, keys=None, data=None):
    return _apply_call(_LIBZENROOM.zenroom_exec, script, conf, keys, data)
```

```
def zenroom_exec(script, conf=None, keys=None, data=None):
    return _apply(_LIBZENROOM.zenroom_exec, script, conf, keys, data)
```

> And also the call argument shouldn't it be function?

`call` is a `callable` hence the name, but it can be changed to function.